src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -343,7 +409,173 @@ public void Stop()
+        /// Otherwise it returns a zero SpanId. 
+        /// </summary>
+        public ref readonly ActivitySpanId SpanId
+        {
+            get
+            {
+                if (!_spanIdSet)
+                {
+                    if (_id != null && IdFormat == ActivityIdFormat.W3C)
+                    {
+                        _spanId = new ActivitySpanId(((ReadOnlySpan<char>)_id).Slice(36, 16));
+                        _spanIdSet = true;
+                    }
+                }
+                return ref _spanId;
+            }
+        }
+
+        /// <summary>
+        /// If the Activity has the W3C format, this returns the ID for the TraceId part of the Id.  
+        /// Otherwise it returns a zero TraceId. 
+        /// </summary>
+        public ref readonly ActivityTraceId TraceId
+        {
+            get
+            {
+                if (!_traceIdSet)
+                {
+                    if (_id != null && IdFormat == ActivityIdFormat.W3C)
+                    {
+                        _traceId = new ActivityTraceId(((ReadOnlySpan<char>)_id).Slice(3, 32));
+                        _traceIdSet = true;
+                    }
+                }
+                return ref _traceId;
+            }
+        }
+
+        /// <summary>
+        /// If the parent Activity ID has the W3C format, this returns the ID for the SpanId part of the ParentId.  
+        /// Otherwise it returns a zero SpanId. 
+        /// </summary>
+        public ref readonly ActivitySpanId ParentSpanId
+        {
+            get
+            {
+                if (!_parentSpanIdSet && _parentId != null && IsW3CId(_parentId))
+                {
+                    _parentSpanId = new ActivitySpanId(((ReadOnlySpan<char>)_parentId).Slice(36, 16));
+                    _parentSpanIdSet = true;
+                }
+                return ref _parentSpanId;
+            }
+        }
+
+        /// <summary>
+        /// Returns the format for the ID.   
+        /// </summary>
+        public ActivityIdFormat IdFormat { get; private set; }
+
+        /* static state (configuration) */
+        /// <summary>
+        /// Activity tries to use the same format for IDs as its parent.
+        /// However if the activity has no parent, it has to do something.   
+        /// This determines the default format we use.  
+        /// </summary>
+        public static ActivityIdFormat DefaultIdFormat
+        {
+            get
+            {
+                if (s_DefaultIdFormat == ActivityIdFormat.Unknown)
+                    s_DefaultIdFormat = ActivityIdFormat.Hierarchical;
+                return s_DefaultIdFormat;
+            }
+            set
+            {
+                if (!(ActivityIdFormat.Hierarchical <= value && value <= ActivityIdFormat.W3C))
+                    throw new ArgumentException($"value must be a valid ActivityIDFormat value");
+                s_DefaultIdFormat = value;
```

Comment:
The resulting race conditions here are ok?

Reply:
I think so.   The expectation is that both DefaultIdFormat and ForceDefaultIdFormat will be set by the logging system before any interesting logging, and not change thereafter  (it is a setup value).    There is an issue that multiple logging systems could 'fight' over the property's value, but the expectation is that that was a configuration error (you should not try to have two logging system that have different ideas about what the ID format it).